netpyne/batch.py

```diff
@@ -44,24 +59,142 @@ def tupleToStr (obj):
     return obj
 
 
+# -------------------------------------------------------------------------------
+# Evolutionary optimization: Parallel evaluation
+# -------------------------------------------------------------------------------
+def evaluator(candidates, args):
+    global ngen
+    ngen += 1
+    total_jobs = 0
+    
+    # read params or set default
+    simDataFolder = args.get('simDataFolder')
+    # read batch params declared in runCfg
+    script = args.get('batch').get('script')
+    numproc = args.get('batch').get('numproc')
+    mpiCmd = args.get('batch').get('mpiCommand')
+    paramNames = args.get('batch').get('paramNames')
+    # fitness function as expression
+    fitness_expression = args.get('fitness')
+    default_fitness = args.get('default_fitness')
+    
+    #run slurm jobs
+    for candidate_index, canditate in enumerate(candidates):
+        # required for slurm
+        sleep(args.get('sleepInterval', 1))
+        jobName = "gen_" + str(ngen) + "_cand_" + str(candidate_index)
+        simDataPath = simDataFolder + '/' + jobName
+        # script to run with mpi
+        command = '%s -np %d nrniv -python -mpi %s netParams=%s simConfig=%s filename=%s ' % (mpiCmd, numproc, script, args.get('netParamPath'), args.get('simConfigPath'), simDataPath)
```

Comment:
This is using the old format for passing parameters to NEURON, i.e. via command line. Now we want to use instead the netpyne format, so there is no need to redefine the code to launch sims via SLURM. You can reuse what is already here: https://github.com/Neurosim-lab/netpyne/blob/development/netpyne/batch.py#L256
The idea would be to make that code a function that is called both for the 'grid' and for the 'evol' optimization algorithms. The type of algorithm (grid or evol) should be independent of the way to run them (e.g. local mpi, hpc torque, hpc slurm, ...)